### ISSUE TITLE
relation query fails when thousands of records

### ISSUE THREAD

**Issue by linojon** [prisma/prisma#11950]
### Bug description

I have a schema that includes
```
model Post {
   ...
   topics Topic[]
}

model Topic {
  id   Int     @id @default(autoincrement())
  name String?

  posts Post[]
  @@map("topics")
}
```

The database has 28 topic records and 50,000 post records. Some topics have over 20,000 posts. Running the query 

```
  const topics = await prisma.topic.findMany({
    orderBy: { name: 'asc' },
    select: {
      id: true,
      name: true,
      posts: true
    }
  })
```
with this (non descript and misleading) error
```
error - Error:
Invalid `prisma.topic.findMany()` invocation:
  Can't reach database server at `localhost`:`5432`
```
Removing `posts: true` is fine. Also this works,
```
  const topics = await prisma.topic.findMany({
    orderBy: { name: 'asc' },
    select: {
      id: true,
      name: true,
      _count: {
        select: { posts: true }
      }
    }
  })
```

I normally do not run this query with `posts: true` but Prisma Studio does. I reported the problem there and they asked i file an issue here instead.

For more details see https://github.com/prisma/studio/issues/853

### How to reproduce

see above

### Expected behavior

see above

### Prisma information

see above

### Environment & setup

- OS: 
- Database: 
- Node.js version: 
Windows 10
Postgres
$ node -v
v14.18.2

### Prisma Version

```

    "@prisma/client": "^3.9.2",

    "prisma": "^3.9.2",

```

**Comment by elarens** [comment]
I'm having the same error with findMany(), just tested it to confirm. When I send a request to get 10,000 results, it works (albeit takes a while), but if it's 200,000, it fails almost immediately with the same error as described above. Same Prisma Client version 3.9.2 (had a similar error with previous clients too)

**Comment by neoromantic** [comment]
Is there any plans to do something with this? Prisma Studio is virtually unusable.

**Comment by jkomyno** [comment]
Closing this in favour of https://github.com/prisma/prisma/issues/8832, please see [this comment](https://github.com/prisma/prisma/issues/8832#issuecomment-1237850099).

**Comment by jkomyno** [comment]
We have closed the original issue in `prisma@4.4.0`, please see [this comment](https://github.com/prisma/prisma/issues/8832#issuecomment-1260533913).
